### PR TITLE
algebra: adding tests for `matrix_bufFree()` function

### PR DESCRIPTION
**Pull request by PNieck**
JIRA: PP-71

## Description

This PR is adding tests for function `matrix_bufFree`

## Motivation and Context

The more tests, the better

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here). `armv7a9-zynq7000-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: algebra/tests/matrix/basic_interactions.c [hunk 1]
```diff
@@ -292,3 +292,49 @@ TEST_GROUP_RUNNER(group_matrix_bufAlloc)
...
+	/* dynMat.data is equal to NULL - see TEST_SETUP */
+	matrix_bufFree(&dynMat);
+
+	TEST_ASSERT_NULL(dynMat.data);
+}
+
+
+TEST(group_matrix_bufFree, matrix_bufFree_settingDataToNULL)
+{
+	TEST_ASSERT_FALSE(matrix_bufAlloc(&dynMat, ROWS, COLS));
```
Comment (niewim19) [inline algebra/tests/matrix/basic_interactions.c:328]: If you rebase this branch you can use proper macro to check value returned by `matrix_bufAlloc()`

File: algebra/tests/matrix/basic_interactions.c [hunk 2]
```diff
@@ -292,3 +292,49 @@ TEST_GROUP_RUNNER(group_matrix_bufAlloc)
...
+TEST(group_matrix_bufFree, matrix_bufFree_passingDataEqualNULL)
+{
+	/* dynMat.data is equal to NULL - see TEST_SETUP */
+	matrix_bufFree(&dynMat);
+
+	TEST_ASSERT_NULL(dynMat.data);
+}
+
+
+TEST(group_matrix_bufFree, matrix_bufFree_settingDataToNULL)
```
Comment (niewim19) [inline algebra/tests/matrix/basic_interactions.c:326]:
```suggestion
TEST(group_matrix_bufFree, matrix_bufFree_normal)
```

File: algebra/tests/matrix/basic_interactions.c [hunk 3]
```diff
@@ -292,3 +292,49 @@ TEST_GROUP_RUNNER(group_matrix_bufAlloc)
...
+}
+
+
+TEST(group_matrix_bufFree, matrix_bufFree_passingDataEqualNULL)
+{
+	/* dynMat.data is equal to NULL - see TEST_SETUP */
+	matrix_bufFree(&dynMat);
+
+	TEST_ASSERT_NULL(dynMat.data);
+}
```
Comment (niewim19) [inline algebra/tests/matrix/basic_interactions.c:323]:
```suggestion
TEST(group_matrix_bufFree, matrix_bufFree_doubleFreeSafe)
{
	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK, matrix_bufAlloc(&dynMat, ROWS, COLS))
	
	matrix_bufFree(&dynMat);
	TEST_ASSERT_NULL(dynMat.data);
	
	matrix_bufFree(&dynMat);
	TEST_ASSERT_NULL(dynMat.data);
}
```
I think this way this test covers more scenarios and is more understandable. (works only after rebase)

**Review by niewim19 — CHANGES_REQUESTED** [review]
Good to see tests even for such trivial functions. Please reffer to given remarks

File: algebra/tests/matrix/basic_interactions.c (hunk 2 above)
Comment (PNieck) [reply algebra/tests/matrix/basic_interactions.c:326]: I will use `matrix_bufFree_std` - same naming convention as with `matrix_zeroes` tests. What do you think?

Comment (niewim19) [reply algebra/tests/matrix/basic_interactions.c:326]: `std` is even better

**Review by niewim19 — APPROVED** [review]
LGTM

**Pull request merged**